aten/src/ATen/native/ComplexHelper.cpp

```diff
@@ -0,0 +1,54 @@
+#include <ATen/native/ComplexHelper.h>
+
+#include <ATen/native/TensorIterator.h>
+
+namespace at {
+namespace native {
+
+DEFINE_DISPATCH(complex_stub);
+DEFINE_DISPATCH(complex_polar_stub);
+
+Tensor& complex_out(Tensor& result, const Tensor& real, const Tensor& imag) {
+  TORCH_CHECK(real.scalar_type() == imag.scalar_type(),
```

Comment:
why should this be necessary? I think we should just do type promotion here

Reply:
removed.